api/solarperformanceinsight_api/compute.py

```diff
@@ -497,19 +497,29 @@ def compare_expected_and_actual(job: models.StoredJob, si: storage.StorageInterf
     save_results_to_db(job.object_id, result_list, si)
 
 
-def _zero_div(a, b):
-    out = a / b
+def _zero_nans(out, a):
     a_almost_zero = abs(a) < 1e-16
     nans = pd.isnull(out)  # nan when 0 / 0,  a > 0 /0 => inf
     out[a_almost_zero & nans] = 0.0
     return out
 
 
+def _zero_div(a, b):
+    out = a / b
+    return _zero_nans(out, a)
+
+
 def _temp_factor(gamma, t_ref, t_actual):
     t0 = 25.0
     return _zero_div(1 - gamma * (t_actual - t0), 1 - gamma * (t_ref - t0))
 
 
+def _inf_mul(a, b):
+    """set 0 * inf = 0"""
+    out = a.mul(b, axis=0)
```

Comment:
out of curiosity, why do you need to use `mul` here but you can use `/` above?

Reply:
`/` happens to operate on pd.Series where `a` in `_inf_mul` is a dataframe and otherwise broadcasts to the wrong axis. Certainly somewhat dangerous and ambiguous; I'll change it a bit